fetchFromGitHub: Always add meta.homepage to the derivation

primeos opened the pull request:
> The attribute meta.homepage was only added if "fetchFromGitHub = false"
which might be unexpected. Now, it will be set unconditionally.

Not sure if this is a good idea / worth it. Consider it an suggestion (could make it more foolproof https://github.com/NixOS/nixpkgs/commit/a33402b89009094d7c7b43d9f3c542120a8b55d2 :smile:).

primeos committed:
fetchFromGitHub: Always add meta.homepage to the derivation

The attribute meta.homepage was only added if "fetchFromGitHub = false"
which might be unexpected. Now, it will be set unconditionally.

orivej commented on pkgs/top-level/all-packages.nix:
```diff
@@ -223,7 +222,8 @@ with pkgs;
       netrcImpureEnvVars = [ "${varBase}USERNAME" "${varBase}PASSWORD" ];
-    } // passthruAttrs) // { inherit rev; };
+    } // passthruAttrs) // { inherit rev; })
+  // { meta.homepage = "${baseUrl}/"; };
```
Could you also extract `passthruAttrs` down here?

Then I'm going to merge this.

primeos replied:
Did some refactoring in 5d0f7dc721997424c85624df0713c07b9cc79fc4 which makes it possible. If I didn't miss anything 5d0f7dc721997424c85624df0713c07b9cc79fc4 shouldn't change the result at all.

The refactoring was necessary to avoid the `passthruAttrs` redundancy as these attributes are required for both `fetchzip` and `fetchgit` (e.g. the `sha256` attribute).

primeos committed:
fetchFromGitHub: Refactor the code

And remove the tailing slash from "meta.homepage" as suggested by
@orivej (because GitHub doesn't add the final slash).

orivej commented on pkgs/top-level/all-packages.nix:
```diff
@@ -198,32 +198,28 @@ with pkgs;
+          netrcImpureEnvVars = [ "${varBase}USERNAME" "${varBase}PASSWORD" ];
+        }))
+      // passthruAttrs // { inherit name; };
+  in fetcher fetcherArgs // { meta.homepage = "${baseUrl}/"; inherit rev; };
```
I like this, thank you! (Although extracting the private section into `privateAttrs` will make it even better.)
I just noticed that GitHub does not add the final slash when you navigate to a project page, so we shouldn't too.

primeos replied:
Good points, thanks! I've changed it accordingly.

orivej reviewed (approved):
I have confirmed that `nox-review pr 29815` rebuilds nothing.
